### PR TITLE
docs: add rainbow animation switcher

### PR DESCRIPTION
**Pull request by userquin**
This PR allows disable the rainbow animation:
- initialize `animate-rainbow` from local storage, using media query (`prefers-reduced-motion: reduce`) for initial value, once loaded it is ignored: check logic in `RainbowAnimtationSwitcher.vue`
- update `rainbow.css`  styles: removing `prefers-reduced-motion: reduce` and using `html:not(.rainbow)` to disable the animation
- added switcher `RainbowAnimtationSwitcher.vue`: using `i-tabler:rainbow` and `i-tabler:rainbow-off`  when animation enabled and disabled repectively
- ~~overriden all required VP components, `VPSwitcher => UnoCSSSwitcher`, `VPNavScreen => UnoCSSNavScreen`, `VPNavBarExtra => UnoCSSNavBarExtra` and `VPNavBar => UnoCSSNavBar`: all VP* components using UnoCSS** prefix, check aliases in `docs/vite.config.ts`: copy/paste replacing local imports with `vitepress/dist/*`~~

Check the following screenshots or just check the PR preview below:

<details>
<summary><i>Desktop</i></summary>

imagen
</details>

<details>
<summary><i>Narrow</i></summary>

imagen
</details>

<details>
<summary><i>Mobile</i></summary>

imagen
</details>

From this discussion: https://github.com/unocss/unocss/discussions/3601

### CODE REVIEW

**Comment by userquin** [comment]
upps, somethong wrong initializing `animate-rainbow`...

File: docs/.vitepress/theme/components/NavScreenRainbowAnimation.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  <div class="NavScreenRainbowAnimtation">
+    <p class="text">
+      Rainbow animation
+    </p>
+    <RainbowAnimationSwitcher />
+  </div>
+</template>
+
+<style scoped>
+.NavScreenRainbowAnimtation {
```
Comment (LeoniePhiline) [inline docs/.vitepress/theme/components/NavScreenRainbowAnimation.vue:15]:
```suggestion
.NavScreenRainbowAnimation {
```

File: docs/.vitepress/theme/components/NavScreenRainbowAnimation.vue [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<script lang="ts" setup>
+import RainbowAnimationSwitcher from './RainbowAnimationSwitcher.vue'
+</script>
+
+<template>
+  <div class="NavScreenRainbowAnimtation">
```
Comment (LeoniePhiline) [inline docs/.vitepress/theme/components/NavScreenRainbowAnimation.vue:6]:
```suggestion
  <div class="NavScreenRainbowAnimation">
```

**Review by LeoniePhiline — COMMENTED** [review]
Lil' typo.

**Comment by antfu** [comment]
I appreciate the detailed care regarding style consistency! But tbh I am a bit worried about the amount of code we vendored - it would be hard for us to keep up the future updates in VP.

For this feature, I don't think it would be need to be treated as dark toggle - as it would be used very often. I would suggest to hide it in a dropdown menu so we don't need to worry about RWD either (which should also minimize the vendor components we need)

**Comment by userquin** [comment]
Moved rainbow switcher to `home-hero-actions-after` and `sidebar-nav-after` slots.

File: docs/components.d.ts [hunk 3]
```diff
@@ -12,7 +12,19 @@ declare module 'vue' {
...
     RouterLink: typeof import('vue-router')['RouterLink']
     RouterView: typeof import('vue-router')['RouterView']
+    ScreenRainbowAnimation: typeof import('./.vitepress/theme/components/ScreenRainbowAnimation.vue')['default']
+    UnoCSSNavBar: typeof import('./.vitepress/theme/components/UnoCSSNavBar.vue')['default']
+    UnoCSSNavBarExtra: typeof import('./.vitepress/theme/components/UnoCSSNavBarExtra.vue')['default']
+    UnoCSSNavScreen: typeof import('./.vitepress/theme/components/UnoCSSNavScreen.vue')['default']
+    UnoCSSRainbow: typeof import('./.vitepress/theme/components/UnoCSSRainbow.vue')['default']
+    UnoCSSScreen: typeof import('./.vitepress/theme/components/UnoCSSScreen.vue')['default']
+    UnoCSSSwitcher: typeof import('./.vitepress/theme/components/UnoCSSSwitcher.vue')['default']
+    UnoCSSSwticher: typeof import('./.vitepress/theme/components/UnoCSSSwticher.vue')['default']
```
Comment (userquin) [inline docs/components.d.ts:28]: cleanup entries

File: docs/.vitepress/theme/index.ts [hunk 4]
```diff
@@ -14,11 +16,12 @@ import 'uno.css'
 let homePageStyle: HTMLStyleElement | undefined
 
 export default {
-  ...Theme,
+  ...DefaultTheme,
```
Comment (userquin) [inline docs/.vitepress/theme/index.ts:19]: use `extends`

File: docs/.vitepress/theme/components/UnoCSSSwitcher.vue [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+  position: absolute;
+  top: 3px;
+  left: 3px;
+  width: 12px;
+  height: 12px;
+  color: var(--vp-c-text-2);
+}
+
+.dark .icon :deep([class^='i-']),
+.dark .icon :deep([class^='vpi-']) {
```
Comment (userquin) [inline docs/.vitepress/theme/components/UnoCSSSwitcher.vue:61]: rename sfc and cleanup vp styles, now only used for rainbow animation switcher

File: docs/.vitepress/theme/components/RainbowAnimationSwitcher.vue [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+.RainbowAnimationSwitcher[aria-checked="true"] .animated {
+  opacity: 1;
+}
+
+.RainbowAnimationSwitcher[aria-checked="false"] :deep(.check) {
+  /*rtl:ignore*/
+  transform: translateX(18px);
+}
+
+html.browser-safari .group,
```
Comment (userquin) [inline docs/.vitepress/theme/components/RainbowAnimationSwitcher.vue:116]: remove safari from here, it seemt the rainbow animation is working

**Review by antfu — APPROVED** [review]

**Pull request merged**